### ISSUE TITLE
Inconsistent constructions with SmallGroups

### ISSUE THREAD

**Issue by mkeilberg** [gap-system/gap#4043]
The following in 4.11.0 (cygwin64 install):

```
G := SmallGroup(5^7,656);;
H := OneSmallGroup(Size,5^7,G->IdGroup(G)[2],656);;
IsomorphismGroups(G,H);
```

returns fail, and they are indeed non-isomorphic groups: G has an irreducible representation of degree 25, whereas H does not.  Similar issues occur when using AllSmallGroups vs SmallGroup to build one or more groups.

**Comment by hulpke** [comment]
[Request aside: Would you be willing to tell us your name and affiliation ?]

There was an issue of inconsistent numbering in degrees p^7 for p=3,5,7,11 between GAP and Magma. This was resolved by using renumbering permutations (though the person who implemented this first used the wrong permutations).

Apparently this same person who messed up the renumbering did not realize that the selection functions (One/AllSmallGroups) need to do a similar translation of the IdGroup number. (When I fixed the numbering last year I did not touch this, as I though it had been taken care of.)

Short story -- if you use One/AllSmallGroups in degrees p^7 for p=3,5,7,11, the IdGroup numbers will be  set wrongly.

I have submitted a PR
https://github.com/gap-packages/smallgrp/pull/52
for the package that should fix this.

**Comment by mkeilberg** [comment]
@hulpke The 'm' in the user name stands for Marc.  I currently have no affiliation.

**Comment by hulpke** [comment]
Thank you. Alexander Hulpke

**Comment by hulpke** [comment]
Let me add that if you have past results that were affected and need number translations, I can give you the permutation needed.

**Comment by fingolfin** [comment]
I believe this was fixed by @hulpke in PR #52 and subsequently in version 1.4.2 of this package. Now `IsomorphismGroups(G,H);` doesn't return `fail` (instead it seems to run for a looong time... I didn't wait to see whether / when it finishes (it shouldn't take so long, really?)

**Comment by hulpke** [comment]
@fingolfin With the code from GAP PR 4219 (https://github.com/gap-system/gap/pull/4219, which incidentally still is waiting for the formal approval so that I can merge), the isomorphism test succeeds on my machine in under 7 minutes. Given that the groups have order 5^7, I think this is actually pretty good.
(There is no special check whether a map generators->generators is an isomorphism, as the chance of this seems to be tiny)

**Comment by wilfwilson** [comment]
Thank you for the initial report @mkeilberg and thanks for your working fixing the SmallGrp package @hulpke, and @fingolfin for releasing v1.4.2 of the package. Since SmallGrp 1.4.2 is now included in the latest GAP release (4.11.1 of March 2021, thank you to everyone who helped with that), I think it's appropriate to close this issue now.

Please reopen if you disagree.

(And the isomorphism test also took under 7 minutes on my computer, too 🙂)
